nftables.go

```diff
@@ -298,7 +299,13 @@ func (nft *realNFTables) List(ctx context.Context, objectType string) ([]string,
 
 // ListRules is part of Interface
 func (nft *realNFTables) ListRules(ctx context.Context, chain string) ([]*Rule, error) {
-	cmd := exec.CommandContext(ctx, nft.path, "--json", "list", "chain", string(nft.family), nft.table, chain)
+	// If no chain is given, return all rules from within the table.
+	var cmd *exec.Cmd
+	if chain == "" {
+		cmd = exec.CommandContext(ctx, nft.path, "--json", "list", "table", string(nft.family), nft.table)
+	} else {
+		cmd = exec.CommandContext(ctx, nft.path, "--json", "list", "chain", string(nft.family), nft.table, chain)
+	}
```

Comment:
this seems a bit of a stretch for an API, since now it list chains and rules based on a special input

I'm not quite familiar with the implementation, @danwinship should know better, but does the  other generic methods in this file `List(ctx context.Context, objectType string) ([]string, error) ` or `ListElements` do not solve the problem?
If not, should we add a new method?